### PR TITLE
Adjust Payout Request API

### PR DESCRIPTION
**Pull request by 2e3s**
## Checklist

- [x] Writing style
- [x] API design standards

### CODE REVIEW

File: openapi/paths/storefront/payout-requests@{id}.yaml [hunk 1]
```diff
@@ -19,10 +19,47 @@ get:
...
+  x-products:
+    - Storefront
+  tags:
+    - Transactions
+  summary: Update a payout request
+  operationId: StorefrontPatchPayoutRequest
+  x-sdk-operation-name: update
+  security:
+    - CustomerJWT: []
+  description: Update a payout request with specified ID.
```
Comment (noon-dawg) [inline openapi/paths/storefront/payout-requests@{id}.yaml:39]:
```suggestion
  description: Update a payout request with a specified ID.
```

File: openapi/paths/payout-requests@{id}.yaml [hunk 2]
```diff
@@ -24,3 +24,41 @@ get:
...
       $ref: ../components/responses/NotFound.yaml
+put:
+  x-products:
+    - Core
+  tags:
+    - Transactions
+  summary: Upsert a payout request
+  operationId: PutPayoutRequest
+  x-sdk-operation-name: update
+  description: Create or update (upsert) a payout request with specified identifier string.
```
Comment (noon-dawg) [inline openapi/paths/payout-requests@{id}.yaml:35]: nitpick: based on style
```suggestion
  description: Creates or updates (upserts) a payout request with specified a ID.
```

File: openapi/components/schemas/PayoutRequest.yaml [hunk 3]
```diff
@@ -37,18 +37,16 @@ properties:
     readOnly: true
     enum:
       - pending
```
Comment (mattsanf) [inline openapi/components/schemas/PayoutRequest.yaml:39]: I think we need another status for when the customer has selected the instrument for the payout request and awaits merchant approval, no?

Comment (2e3s) [reply openapi/components/schemas/PayoutRequest.yaml:39]: I've added [instrument-selected status](https://github.com/Rebilly/api-definitions/pull/1212/commits/96d7b5e08ff3231763dee5f6db01606ce768578f)

**Review by mattsanf — APPROVED** [review]

**Review by ninjasimon — APPROVED** [review]

**Review by noon-dawg — APPROVED** [review]

**Pull request merged**